scripts/darwin/metrics.sh

```diff
@@ -30,4 +47,99 @@ else
   target_branch=$(echo "$data" | grep "Target branch is:" | awk '{print $4}')
 fi
 
+# Pathenon Metrics Application Analysis
+#
+# ********* Overview ********
+#
+# The below command at the bottome of the page s rather involved. Here the parthenon
+# metrics app is being told to analyze the results of the parthenon performance
+# regression tests. This consists of several steps:
+#
+# 1. performance test data must exist in the local parthenon repository
+#
+# 2. A comparison between the local test results and previous results must be conducted
+#    so that a regression can be spotted.
+#
+#      * In the case of a pull request, this means comparing the local results with results
+#        of the target branch.
+#
+#      * In the case of a scheduled run, where there is no merge, the local test results
+#        which are assumed to be the most up to date are compared with the performance of
+#        previous commits on the same branch.
+#
+#    The performance metrics of the non local repository are stored in the parthenon wiki.
+#    The metrics app will download the wiki and read the performance metrics from the
+#    appropriate files in the wiki to make the comparison.
+#
+# 3. Then the metrics app will generate figures displaying performance data (regressions should be easy to spot).
+#
+# 4. Next a wiki page file will be generated in the wiki with links to the figures. This
+#    wiki page file will then be uploaded to the parthenon wiki.
+#
+# 5. The figures will be uploaded to an empty figures branch (orphan branch) in the parthenon repository:
+#    https://github.com/lanl/parthenon/tree/figures
+#
+#    It may be necessary in the future to squash the commit history in that branch occassionally.
+#
+# 6. Finally, a status report will be sent to github and associated with commit of the current branch,
+#    this status should show up next to other status reports such as Codacy, Deepcode and gitlab ci.
+#    The status will contain a link to the wiki page where the performance metrics are posted.
+#
+# NOTE a. Why are we uploading figures to their own branch in the parthenon repository and not
+#      simply storing them in parthenon.wiki?
+#
+#      Binary files will bloat the wiki repository
+#
+#      b. Why not simply place the image files in a branch on wiki?
+#
+#      I could not figure out how to link to the images that are located in a branch
+#      in the wiki repository. from the wiki page.
+#
+#      c. Why not store them in develop in the partheon repo?
+#
+#      Again we don't want to bloat the partheon repo with binary files.
+#
+# ********** Flags and thier meanings **********
+#
+# -p path_to_matrics_application_permissions_file
+#
+# The path to the .pem file so the metrics application can authenticate with the github server.
```

Comment:
```suggestion
# The path to the .pem file so the metrics application can authenticate with the GitHub server.
```
May need to capitalize "GitHub" in other places too.

Would be helpful to have a brief note here on how to obtain the permissions file.